miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultStudyService.java

```diff
@@ -0,0 +1,93 @@
+package uk.ac.bbsrc.tgac.miso.service.impl;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import uk.ac.bbsrc.tgac.miso.core.data.Study;
+import uk.ac.bbsrc.tgac.miso.core.data.StudyType;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.StudyImpl;
+import uk.ac.bbsrc.tgac.miso.core.store.StudyStore;
+import uk.ac.bbsrc.tgac.miso.service.StudyService;
+import uk.ac.bbsrc.tgac.miso.service.security.AuthorizationManager;
+
+@Transactional(rollbackFor = Exception.class)
+@Service
+public class DefaultStudyService implements StudyService {
+  @Autowired
+  private AuthorizationManager authorizationManager;
+
+  @Autowired
+  private StudyStore studyStore;
+
+  @Override
+  public void delete(Study study) throws IOException {
+    authorizationManager.throwIfNonAdmin();
+    studyStore.remove(study);
+  }
+
+  @Override
+  public Collection<Study> get() throws IOException {
+    return authorizationManager.filterUnreadable(studyStore.listAll());
+  }
+
+  @Override
+  public Study get(long studyId) throws IOException {
+    Study s = studyStore.get(studyId);
+    authorizationManager.throwIfNotReadable(s);
+    return s;
+  }
+
+  @Override
+  public Collection<Study> getByProjectId(long projectId) throws IOException {
+    return authorizationManager.filterUnreadable(studyStore.listByProjectId(projectId));
+  }
+
+  @Override
+  public Collection<Study> getBySearch(String query) throws IOException {
+    return authorizationManager.filterUnreadable(studyStore.listBySearch(query));
+  }
+
+  @Override
+  public Map<String, Integer> getColumnSizes() throws IOException {
+    return studyStore.getStudyColumnSizes();
+  }
+
+  @Override
+  public StudyType getType(long id) {
+    return studyStore.getType(id);
+  }
+
+  @Override
+  public Collection<StudyType> getTypes() throws IOException {
+    return studyStore.listAllStudyTypes();
+  }
+
+  @Override
+  public Collection<Study> getWithLimit(long limit) throws IOException {
+    return authorizationManager.filterUnreadable(studyStore.listAllWithLimit(limit));
+  }
+
+  @Override
+  public long save(Study study) throws IOException {
+    if (study.getId() == StudyImpl.UNSAVED_ID) {
+      study.setLastModifier(authorizationManager.getCurrentUser());
+      return studyStore.save(study);
+    } else {
+      Study original = studyStore.get(study.getId());
+      authorizationManager.throwIfNotWritable(original);
+      original.setAccession(study.getAccession());
+      original.setAlias(study.getAlias());
+      original.setDescription(study.getDescription());
+      original.setLastModifier(authorizationManager.getCurrentUser());
+      // project is immutable
+      original.setStudyType(studyStore.getType(original.getStudyType().getId()));
```

Comment:
What use cases are you thinking for wanting StudyType to be immutable? (I see it as being basically an enum, and distinct from Project, so  I can't yet imagine a case where changing it would matter much)